csrc/id_model/schedule.cpp

```diff
@@ -10,10 +10,48 @@
 
 namespace nvfuser {
 
+IterDomain* representativeId(const ValGroup& vg) {
+  IterDomain* rep = nullptr;
+
+  auto preferNewIterType = [&rep](IterDomain* new_id) {
+    if (rep->isReduction()) {
+      return new_id->isBroadcast() || new_id->isIteration();
+    } else if (rep->isBroadcast()) {
+      return new_id->isIteration();
+    } else if (rep->isIteration()) {
+      return false;
+    } else {
+      // Prefer anything else to a non-iter, non-bcast, non-reduction ID
+      return true;
+    }
+  };
+
+  auto preferNewExtent = [&rep](IterDomain* new_id) {
```

Comment:
Unless this graph is Exact, loop promotion man need to be used as well, but I'm not really sure which graph is intended to be used. Maybe only the Exact graph?

Reply:
For AbstractTensor scheduling, we'll use something similar to Exact, but with concretized broadcasts mapped. I don't think we need to consider loop promotion though since we are just looking for a concrete IterDomain to use in an `IterDomain::merge` operation, for example.

Reply:
Hmm, actually, why do we need this? I understand the preference of IterTypes. Why does the extent also matter?

Reply:
It probably doesn't matter much and I might have been overthinking things. I was thinking for example that when we merge ValGroups, we create a new output IterDomain that we can then potentially map other IterDomains into. Assuming those mapped IterDomains have symbolic extents, we would prefer that the original merged ID has a constant extent so that the ValGroup has a constant extent in at least one of its members. If we used representatives with symbolic extents when some constant ones exist, then that would not be the case: we'd have all symbolic extents in the merged ValGroup.

Reply:
My brain is not working today. Could you give a concrete example?

Reply:
Suppose we have a Fusion with the following TVs (among others):

```
tv3[ i5, i6 ]
tv4[ 3, 4 ]
tv5 = tv3 + tv4
```
So `i5` is mapped to 3 and `i6` is mapped to 4.
```
AbstractTensor at = [ {i5, 3, ...}  {i6, 4, ...} ]
at.merge(0)
```
This merge step is done by using representative IDs for both `ValGroup`s and merging them to form a new ID, and then edges are added to the `ValGraph` representing the merge. If we use the first IDs in each group instead of the ones with constant extent, then the new `ValGroup` contains `{ i5*i6 }` instead of `{ 12 }`.

Now we apply this to `tv3`, but not `tv4`. This means the loop domain of `tv3` becomes `i5*i6`. We don't currently re-use any of the graphs we build for further steps in scheduling/lowering, but _if we did_ then we would already have a "nice" ID mapped to that loop domain, so we could infer that its extent is actually 12.

Note that this wouldn't have any effect right now since we create new IdModel during lowering anyway, so I'm fine with removing this part of the PR to improve clarity if you would like me to.

Reply:
Oh, somehow I was thinking about AbstractTensor with IterDomains, not ValGroups, so when you have a merge like:

```
AbstractTensor at = [ {i5, 3, ...}  {i6, 4, ...} ]
at.merge(0)
```

I was expecting to get:

```
[{i5 * i6}, {3 * 4}, ...]
```

Reply:
Now I understand that extents do matter. I still don't know what this would mean with any graph except for Exact.

In Permissive, a ValGroup may have IDs with different extents, and all of them may be non-broadcast. What is representative then? 

I feel that this would be only well defined for the Exact graph, where extents are all guaranteed to be the same. If that's the case, choosing which extent to use shouldn't matter.

Reply:
> I feel that this would be only well defined for the Exact graph, where extents are all guaranteed to be the same. If that's the case, choosing which extent to use shouldn't matter.

Good point. I think I will just remove all the extent stuff so that the code is simply choosing based on IterType.

Reply:
Note that what loop promotion tries to analyze is exactly this problem, so that's why I wondered if loop promotion should also be used here.

I think it's also fine to keep this almost as is, as long as some safeguards are added. For example, if two different symbolic extents are given, we could simply throw an error. Depending on what you're planning to use this, that may be good enough to move forward for now. If that's the case, I'd be happy to give a stamp.

Reply:
> Depending on what you're planning to use this, that may be good enough to move forward for now.

All I actually need from this PR is to be sure that it's possible to merge two ValGroups even if they include a mix of Reduction and Iteration domains, which the `IterType` checks will ensure. I think we should probably wait for more motivation to implement extent checks/preferences.